tooling/nargo/src/ops/foreign_calls.rs

```diff
@@ -94,11 +95,22 @@ pub struct DefaultForeignCallExecutor {
     mocked_responses: Vec<MockedCall>,
     /// Whether to print [`ForeignCall::Print`] output.
     show_output: bool,
+    /// JSON RPC client to resolve foreign calls
+    external_resolver: Option<Client>,
 }
 
 impl DefaultForeignCallExecutor {
-    pub fn new(show_output: bool) -> Self {
-        DefaultForeignCallExecutor { show_output, ..DefaultForeignCallExecutor::default() }
+    pub fn new(show_output: bool, resolver_url: Option<&str>) -> Self {
+        let oracle_resolver = resolver_url.map(|resolver_url| {
+            let transport_builder =
+                Builder::new().url(resolver_url).expect("Invalid oracle resolver URL");
```

Comment:
I think that if the user passes in an invalid oracle url, we likely don't want to panic since its a user error -- can you open up an issue for this? It's more of a nit and not a blocker, given that it allows us to avoid returning a result

Reply:
https://github.com/noir-lang/noir/issues/3904